great_expectations/zep/postgres_datasource.py

```diff
@@ -139,18 +122,16 @@ def add_table_asset(self, name: str, table_name: str) -> TableAsset:
         Returns:
             The TableAsset that is added to the datasource.
         """
-        asset = TableAsset(name=name, datasource=self, table_name=table_name)
+        asset = TableAsset(name=name, table_name=table_name)
+        # TODO (kilo59): custom init for `DataAsset` to accept datasource in constructor?
+        # Will most DataAssets require a `Datasource` attribute?
+        asset._datasource = self  # type: ignore[misc] # setter defined on super class
         self.assets[name] = asset
         return asset
 
     def get_asset(self, asset_name: str) -> TableAsset:
         """Returns the TableAsset referred to by name"""
-        try:
-            return self.assets[asset_name]
-        except KeyError as e:
-            raise PostgresDatasourceError(
-                f"No table asset named {asset_name}. Available assets are {self.assets.keys()}"
-            ) from e
```

Comment:
I moved this logic to the super() class `.get_asset` but if you think it would be better to raise a specific `PostgresDatasourceError` here I can catch and re-raise as this error here.